modified_cdc_protocol/operation_types/rehydratepatientsample/protocol.rb

```diff
@@ -0,0 +1,171 @@
+# typed: false
+# frozen_string_literal: true
+
+needs 'Standard Libs/PlanParams'
+needs 'Standard Libs/Debug'
+needs 'Standard Libs/Pipettors'
+needs 'Standard Libs/Units'
+needs 'Standard Libs/AssociationManagement'
+
+needs 'Collection Management/CollectionDisplay'
+needs 'Collection Management/CollectionTransfer'
+
+needs 'Diagnostic RT-qPCR/DiagnosticRTqPCRHelper'
+
+needs 'Tube Rack/TubeRack'
+
+needs 'Modified CDC Protocol/RehydratePatientSampleDebug'
+
+TEMPLATE = 'Template'
+MEDIA = 'Media'
+
+EPP_LABEL = 'Eppendorf Tube'
+
+class Protocol
+  # Standard Libs
+  include PlanParams
+  include Debug
+  include Pipettors
+  include Units
+  include AssociationManagement
+  include PartProvenance
+
+  #Collection Management
+  include CollectionDisplay
+  include CollectionTransfer
+
+  #Diagnostic RT-qPCR
+  include DiagnosticRTqPCRHelper
+
+  #TubeRack
+  include TubeRackHelper
+
+  #Modified CDC Protocol
+  include RehydratePatientSampleDebug
+
+  attr_reader :job_params, :operation_params, :media_volume
+  def default_operation_params
+    {
+      media_volume: 300,
+      media_units: MICROLITERS,
+    }
+  end
+
+  def default_job_params
+    {
+    }
+  end
+
+  def main
+    setup_test(operations)
+    @job_params = update_all_params(
+      operations: operations,
+      default_job_params: default_job_params,
+      default_operation_params: default_operation_params
+    )
+
+    rnase_warning
+
+    safety_warning
+
+    operations.retrieve.make
+
+    operations.each do |op|
+      operation_params = op.temporary[:options]
+      media_volume = {
+        qty: operation_params[:media_volume],
+        units: operation_params[:media_units]
+      }
+
+      op.pass(TEMPLATE)
+
+      item_list = op.input_array(TEMPLATE).map do |fv|
+        fv.item.collection? ? fv.collection : fv.item
```

Comment:
An array input of collections seems a little weird. Do you mean parts?

Reply:
I have been debating about this.   Patient samples could come in two forms:
1. Individual samples (Items)
2. In a stripwell all connected as a collection

In case 2 I thought it would be best to pass the whole collection and perform the operation on all the parts inside that collection.  This assumption made the protocol a bit easier to write but I am not sure if this is a safe assumption...

The stripwells of samples will primarily only ever be used during LOD testing of the workflow and not for any actual patient samples.

Reply:
Fair enough.